topi/python/topi/vision/nms.py

```diff
@@ -289,8 +339,136 @@ def hybrid_nms(data, sorted_index, valid_count,
     return output, box_indices
 
 
+@hybrid.script
+def hybrid_dynamic_nms(data, sorted_index, max_output_size, score_threshold,
```

Comment:
I think we can merge this with existing nms routing. Also we need valid_count to reduce the number of iteration to prevent nms to be the performance bottleneck. For tensorflow nms, since we want the original index of all boxes, we might want to get a mapping from altered box indices to original box indices, in get_valid_counts function.